experimental/QuantumAnalogs/src/QuantumAnalogs.jl

```diff
@@ -0,0 +1,295 @@
+################################################################################
+# Quantum analogs
+#
+# Copyright (C) 2020 Ulrich Thiel, ulthiel.com/math
+#
+# Mar 2023: Imported into OSCAR by UT
+#
+# Future features could include:
+# - Symmetric quantum numbers
+# - Two-colored quantum numbers
+################################################################################
+
+export quantum_integer, quantum_factorial, quantum_binomial
+
+################################################################################
+# Quantum integers
+################################################################################
+@doc raw"""
+    quantum_integer(n::IntegerUnion, q::RingElem)
+    quantum_integer(n::IntegerUnion, q::Integer)
+    quantum_integer(n::IntegerUnion)
+    
+Let ``n ∈ ℤ`` and let ``ℚ(𝐪)`` be the fraction field of the polynomial ring ``ℤ[𝐪]`` in
+one variable ``𝐪``. The **quantum integer** ``[n]_𝐪 ∈ ℚ(𝐪)`` of ``n`` is defined as
+```math
+[n]_𝐪 ≔ \frac{𝐪^n-1}{𝐪-1} \;.
+```
+We have
+```math
+[n]_𝐪 = \sum_{i=0}^{n-1} 𝐪^i ∈ ℤ[𝐪] \quad \text{if } n ≥ 0
+```
+and
+```math
+[n]_𝐪 = -𝐪^{n} [-n]_𝐪 \quad \text{for any } n ∈ ℤ \;,
+```
+hence
+```math
+[n]_𝐪 = - \sum_{i=0}^{-n-1} 𝐪^{n+i} ∈ ℤ[𝐪^{-1}] \quad \text{ if } n < 0 \;.
+```
+This shows in particular that actually
+```math
+[n]_𝐪 ∈ ℤ[𝐪,𝐪^{-1}] ⊂ ℚ(𝐪) \quad \text{ for any } n ∈ ℤ \;.
+```
+Now, for an element ``q`` of a ring ``R`` we define ``[n]_q ∈ R`` as the specialization of
+``[n]_𝐪`` in ``q`` using the two equations above—assuming that ``q`` is invertible in ``R``
+if ``n<0``. Note that for ``q=1`` we obtain
+```math
+[n]_1 = n \quad \text{for any } n ∈ ℤ \;,
+```
+so the quantum integers are "deformations" of the usual integers.
+
+# Functions
+* `quantum_integer(n::IntegerUnion,q::RingElem)` returns ``[n]_q`` as an element of ``R``,
+  where ``R`` is the parent ring of ``q``. 
+* `quantum_integer(n::IntegerUnion,q::Integer)` returns ``[n]_q``. Here, if ``n >= 0`` or
+  ``q = ± 1``, then ``q`` is considered as an element of ``ℤ``, otherwise it is taken as an
+  element of ``ℚ``.
+* `quantum_integer(n::IntegerUnion)` returns ``[n]_𝐪`` as an element of ``ℤ[𝐪^{-1}]``.
+
+# Examples
+```jldoctest
+julia> quantum_integer(3)
+q^2 + q + 1
+
+julia> quantum_integer(-3)
+-q^-1 - q^-2 - q^-3
+
+julia> quantum_integer(3,2)
+7
+
+julia> quantum_integer(-3,2)
+-7//8
+
+julia> K,i = cyclotomic_field(4, "i");
+
+julia> quantum_integer(3, i)
+i
+```
+
+# References
+1. [Con00](@cite)
+2. [KC02](@cite)
+"""
+function quantum_integer(n::IntegerUnion, q::RingElem)
+
+  R = parent(q)
+  if isone(q)
+    return R(n)
+  else
+    z = zero(R)
+    if n >= 0
+      for i = 0:n-1
+        z += q^i
+      end
+    else
+      for i = 0:-n-1
+        z -= q^(n+i)
```

Comment:
```suggestion
      for i = n:-1
        z -= q^i
```

Reply:
OK while this is IMHO slghtly easier for understanding what the code does, the current code matches the explanation in the docstring, so better to leave it the way it is.